discretize/TreeMesh.py

```diff
@@ -170,7 +170,7 @@ def __dirty__(self, val):
             '_faceDiv', '_edgeCurl', '_nodalGrad',
             '_aveFx2CC', '_aveFy2CC', '_aveFz2CC', '_aveF2CC', '_aveF2CCV',
             '_aveEx2CC', '_aveEy2CC', '_aveEz2CC', '_aveE2CC', '_aveE2CCV',
-            '_aveN2CC',
+            '_aveN2CC', '_h_gridded'
```

Comment:
@dccowan, @rowanc1: this now makes sure that `_h_gridded` is cleared on mesh updates. There is also a test.